sqlglot/parser.py

```diff
@@ -456,23 +456,31 @@ def _parse_create(self):
         replace = self._match(TokenType.OR) and self._match(TokenType.REPLACE)
         temporary = self._match(TokenType.TEMPORARY)
 
-        create_token = self._match_set((TokenType.TABLE, TokenType.VIEW)) and self._prev
+        create_token = (
+            self._match_set((TokenType.TABLE, TokenType.VIEW, TokenType.FUNCTION))
+            and self._prev
+        )
 
         if not create_token:
-            self.raise_error("Expected TABLE or View")
+            self.raise_error("Expected TABLE, VIEW, or FUNCTION")
 
         exists = self._parse_exists(not_=True)
-        this = self._parse_table(schema=True)
+        this = None
         expression = None
         properties = None
 
-        if create_token.token_type == TokenType.TABLE:
+        if create_token.token_type == TokenType.FUNCTION:
+            this = self._parse_id_var()
+            if self._match(TokenType.ALIAS):
```

Comment:
is the AS required?

Reply:
Yes, see https://spark.apache.org/docs/3.0.0-preview/sql-ref-syntax-ddl-create-function.html